aws_lambda_powertools/metrics/provider/base.py

```diff
@@ -104,11 +190,14 @@ def handler(event, context):
                 self.log_metrics,
                 capture_cold_start_metric=capture_cold_start_metric,
                 raise_on_empty_metrics=raise_on_empty_metrics,
+                default_dimensions=default_dimensions,
             )
 
         @functools.wraps(lambda_handler)
         def decorate(event, context):
             try:
+                if default_dimensions:
+                    self.set_default_dimensions(**default_dimensions)
```

Comment:
this method doesn't exist in Base. What you'd want to do is keep everything but the logic for `default_dimensions`.

Then override `log_metrics` in `AmazonCloudWatchEMFProvider` to set the default dimensions if available, then call `super().log_metrics(...)`

Reply:
The reason I kept it in the base class is because we are subclassing, so we access the `default_dimensions` of the child class. But yes, it makes a lot of sense to keep provider specializations in their specific class.

Fixed.